### PR TITLE
add google analytics polyfill, fixes brave/brave-browser#4402

### PR DESCRIPTION
**Pull request by snyderp**
Fixes: brave/brave-browser#4402

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [X] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [X] Ran `git rebase master` (if needed).
- [X] Ran `git rebase -i` to squash commits (if needed).
- [X] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: app/brave_generated_resources.grd [hunk 1]
```diff
@@ -417,6 +417,7 @@ By installing this extension, you are agreeing to the Google Widevine Terms of U
       </message>
     </messages>
     <includes>
+      <include name="IDR_BRAVE_GOOGLE_ANALYTICS_POLYFILL" file="resources/js/google_analytics_polyfill.js" type="BINDATA" />
```
Comment (bridiver) [inline app/brave_generated_resources.grd]: @bbondy do you think it's ok to put it in this directory or should we create a new one for things that aren't copied from chrome?

Comment (bbondy) [reply app/brave_generated_resources.grd]: I think it's fine

File: browser/net/brave_ad_block_tp_network_delegate_helper.cc [hunk 2]
```diff
@@ -30,14 +30,26 @@ using content::ResourceType;
...
+}
+
 std::string GetGoogleTagManagerPolyfillJS() {
   static std::string base64_output;
   if (base64_output.length() != 0)  {
     return base64_output;
   }
   std::string str = ui::ResourceBundle::GetSharedInstance().GetRawDataResource(
       IDR_BRAVE_TAG_MANAGER_POLYFILL).as_string();
-  base64_output.reserve(180);
```
Comment (bbondy) [inline browser/net/brave_ad_block_tp_network_delegate_helper.cc]: these were added as an optimization so the buffer doesn't need to reallocate itself multiple times btw. But it's a micro-optimization so it's ok to get rid of it.

File: browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc [hunk 3]
```diff
@@ -116,31 +116,40 @@ TEST_F(BraveAdBlockTPNetworkDelegateHelperTest, Blocking) {
...
+  ASSERT_FALSE(GetPolyfillForAdBlock(true, true, tab_origin, google_analytics_url, &out_url_spec));
   // Shields up, allow ads, tag manager should NOT get polyfill
   ASSERT_FALSE(GetPolyfillForAdBlock(true, true, tab_origin, tag_manager_url, &out_url_spec));
   // Shields up, allow ads, tag services should NOT get polyfill
   ASSERT_FALSE(GetPolyfillForAdBlock(true, true, tab_origin, tag_services_url, &out_url_spec));
   // Shields up, allow ads, normal URL should NOT get polyfill
   ASSERT_FALSE(GetPolyfillForAdBlock(true, true, tab_origin, normal_url, &out_url_spec));
 
+  // Shields down, allow ads, google analytics should get polyfill
+  ASSERT_FALSE(GetPolyfillForAdBlock(false, true, tab_origin, google_analytics_url, &out_url_spec));
```
Comment (bsclifton) [inline browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc]: is this correct? shields down... we still want to polyfill? (or typo in comment)

Comment (snyderp) [reply browser/net/brave_ad_block_tp_network_delegate_helper_unittest.cc]: ah, yuck, typo.  Repushed and edd940b fixes

**Comment by snyderp** [comment]
From Slack conversation with @bsclifton, going to merge this in, since the issue seems to be in CI, not in the code (after several reviews).  Merging and uplifting

**Comment by kjozwiak** [comment]
Reproduced the original issue by loading both `theverge.com` or `overclockers.co.uk` using `0.68.60 Chromium: 75.0.3770.90`. Verification **PASSED** on `macOS 10.14.5 x64` using the following build:

Brave | 0.68.64 Chromium: 75.0.3770.100 (Official Build) nightly (64-bit)
--- | --
Revision | cd0b15c8b6a4e70c44e27f35c37a4029bad3e3b0-refs/branch-heads/3770@{#1033}
OS | Mac OS X

You can definitely see a difference with the fix. Both websites mentioned above load a lot faster compared to builds without the fix.

**Pull request merged**